### ISSUE TITLE
bot commands issues

### ISSUE THREAD

**Issue by panpansh** [malwaredllc/byob#182]
(amd64 build debian) Both client and server 127.0.0.1:10000
-------------------------
Using wget file :
Client
```shell
[...]
DEBUG:__main__:wget error: global name 'urlretrieve' is not defined
DEBUG:__main__:recv_task error: invalid header length
DEBUG:__main__:Connection timed out
$(killed)
```
Server
```shell
 [ 0 @ /home/panpansh]>wget http://bhs.proof.ovh.net/files/1Gio.dat
[ 0 @ /home/panpansh]>
```
-------------------------
Using portscan 127.0.0.1
Client
```shell
[...]
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: portscanner
DEBUG:__main__:Path: None
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition... 
INFO:__main__:[+] Module/Package 'portscanner' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading portscanner...
DEBUG:__main__:Trying to import 'portscanner' as package from: 'http://127.0.0.1:10001/portscanner/__init__.py'
INFO:__main__:[-] 'portscanner' is not a package (HTTP Error 404: File not found)
DEBUG:__main__:[+] Trying to import 'portscanner' as module from: 'http://127.0.0.1:10001/portscanner.py'
DEBUG:__main__:[+] Importing 'portscanner'
DEBUG:__main__:[+] Ready to execute 'portscanner' code
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: util
DEBUG:__main__:Path: None
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition... 
INFO:__main__:[+] Module/Package 'util' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading util...
DEBUG:__main__:Trying to import 'util' as package from: 'http://127.0.0.1:10001/util/__init__.py'
INFO:__main__:[-] 'util' is not a package (HTTP Error 404: File not found)
DEBUG:__main__:[+] Trying to import 'util' as module from: 'http://127.0.0.1:10001/util.py'
DEBUG:__main__:[+] Importing 'util'
DEBUG:__main__:[+] Ready to execute 'util' code
INFO:__main__:[+] 'util' imported succesfully!
INFO:__main__:[+] 'portscanner' imported succesfully!
DEBUG:__main__:[+] portscanner remotely imported
```
Server
```shell
 [ 0 @ /home/panpansh]>portscan 127.0.0.1

^C^C^C^C^C

^C^C
```
-------------------------
"id" present on "sessions" but not set on "clients"
```shell
[panpansh @ /home/panpansh]>clients
   username                    panpansh
   administrator               False
   uid                         4583929b48cb120ecb19076048541416
   sessions                    True
   mac_address                 (removed)
   local_ip                    127.0.1.1
   joined                      2020-02-14 03:50:45.578012
   last_online                 2020-02-14 04:32:47.618971
   public_ip                   (removed)
   platform                    linux2
   architecture                64
   online                      False
   device                      (removed)
   id                          None

[panpansh @ /home/panpansh]>sessions
0
  username           panpansh
  administrator      False
  sessions           True
  uid                4583929b48cb120ecb19076048541416
  local_ip           127.0.1.1
  online             True
  joined             2020-02-14 03:50:45.578012
  last_online        2020-02-14 03:50:45.578129
  public_ip          (removed)
  platform           linux2
  architecture       64
  mac_address        (removed)
  device             (removed)
```

**Comment by panpansh** [comment]
(oops I make a try starting server from venv)

**Comment by panpansh** [comment]
![byob-portscan-stuck](https://user-images.githubusercontent.com/790586/74564203-cea2e400-4f6e-11ea-8be3-4303b4e7fb7c.PNG)
I'm really stuck here :/

**Comment by malwaredllc** [comment]
The portscanner module has a bug - sorry about that. I've fixed it and will push the changes to production sometime this weekend after I do some testing.

**Comment by malwaredllc** [comment]
I will look into the wget module as well - it looks like some changes made to support Python 3 may have broken it for Python 2.

**Comment by panpansh** [comment]
Thank you for taking the time to answer me. You have a very clean tool.
If you allow me, I would like to take this opportunity to list the various problems here.
Here is the log for keylogger:
![image](https://user-images.githubusercontent.com/790586/74568646-bafc7b00-4f78-11ea-8c05-9a198889d7b6.png)

**Comment by malwaredllc** [comment]
@panpansh Thanks for bringing this issues to my attention.

1. **Keylogger:** This module works on Windows (using the PyHook library). However, I haven't been able to get this module to work on Linux (or Unix based systems like Mac OSX), as the analogous library pyxhook either doesn't work or the built-in security of the operating system prevents it from working.
2. **Process:** The `process list` command works for me - however, please note that the process blocking feature (`process block <name>`) only works on Windows as it uses PowerShell scripts.
3. **Screenshot:** This module works reliably for me on all operating systems when running on a physical computer. However, I believe the underlying library I used for screenshots (mss) doesn't work properly on virtual machines.
4. **Upload**: Yes I think organizing uploads by client UID is a good idea.

**Comment by malwaredllc** [comment]
@panpansh Various bugs in modules were fixed in #191 including the issues outlined here.
